### PR TITLE
Add tests for CE dispatching on NATS

### PR DESCRIPTION
**Pull request by radufa**
**Description**

Changes proposed in this pull request:

- Add tests for CE dispatching of messaging based on CE-SDK
- Update go.mod to use the same versions as for event-publisher-proxy

**Related issue(s)**

### CODE REVIEW

File: components/eventing-controller/go.mod [hunk 1]
```diff
@@ -4,13 +4,15 @@ go 1.14
 
 require (
 	github.com/avast/retry-go v2.4.1+incompatible
+	github.com/cloudevents/sdk-go/protocol/nats/v2 v2.3.1
```
Comment (marcobebway) [inline components/eventing-controller/go.mod:7]: Why not use the same version used here https://github.com/kyma-project/kyma/blob/release-1.24/components/eventing-controller/go.mod#L7?

**Comment by marcobebway** [comment]
Why Event publisher was not updated as well?

Ideally it should look like https://github.com/kyma-project/kyma/blob/release-1.24/components/event-publisher-proxy/go.mod#L7.

**Comment by marcobebway** [comment]
The Event publisher was updated on release-1.24 (see [here](https://github.com/kyma-project/kyma/blob/release-1.24/components/event-publisher-proxy/go.mod#L7)), why this is not included in this PR as well?

File: components/eventing-controller/go.mod (hunk 1 above)
Comment (radufa) [reply components/eventing-controller/go.mod:7]: They are different things:
''github.com/cloudevents/sdk-go/v2 v2.4.1"
and 
"github.com/cloudevents/sdk-go/protocol/nats/v2 v2.3.1"

Comment (radufa) [reply components/eventing-controller/go.mod:7]: "github.com/cloudevents/sdk-go/v2 v2.4.1" is used in both event-publisher-proxy and eventing-controller

**Comment by radufa** [comment]
"github.com/cloudevents/sdk-go/v2 v2.4.1" is used in both event-publisher-proxy and eventing-controller on both branches:
- main: 
  https://github.com/kyma-project/kyma/blob/main/components/event-publisher-proxy/go.mod#L7
  https://github.com/kyma-project/kyma/blob/main/components/eventing-controller/go.mod#L7
- release-1.24:
  https://github.com/kyma-project/kyma/blob/release-1.24/components/event-publisher-proxy/go.mod#L7
  https://github.com/kyma-project/kyma/blob/release-1.24/components/eventing-controller/go.mod#L7

**Review by pxsalehi — APPROVED** [review]

**Pull request merged**